symfony/framework-bundle/5.1/src/Kernel.php

```diff
@@ -3,52 +3,26 @@
 namespace App;
 
 use Symfony\Bundle\FrameworkBundle\Kernel\MicroKernelTrait;
-use Symfony\Component\Config\Loader\LoaderInterface;
-use Symfony\Component\Config\Resource\FileResource;
-use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
 use Symfony\Component\HttpKernel\Kernel as BaseKernel;
 use Symfony\Component\Routing\Loader\Configurator\RoutingConfigurator;
 
 class Kernel extends BaseKernel
 {
     use MicroKernelTrait;
 
-    private const CONFIG_EXTS = '.{php,xml,yaml,yml}';
-
-    public function registerBundles(): iterable
+    protected function configureContainer(ContainerConfigurator $container): void
     {
-        $contents = require $this->getProjectDir().'/config/bundles.php';
-        foreach ($contents as $class => $envs) {
-            if ($envs[$this->environment] ?? $envs['all'] ?? false) {
-                yield new $class();
-            }
-        }
+        $container->import('../config/{packages}/*.yaml');
+        $container->import('../config/{packages}/'.$this->environment.'/*.yaml');
+        $container->import('../config/{services}.yaml');
+        $container->import('../config/{services}_'.$this->environment.'.yaml');
```

Comment:
I worked on that part yesterday on a big project.

Why do you need the accolade? By using it, you force the use of Symfony/Finder over the Spl implementation and the `uasort(SortbaleIterator)` is wayyyyyyy much slower that the simpler `uasort(strnatcmp)`. 

If you remove the multiple extensions, you do not need Symfony/Finder at all, and a nice perf improvement. 

The only experct is that it doesn't handle the file/direction absence, which is not supposed to happen..

Reply:
(perf in dev mode only)

Reply:
For the last two lines, the brackets make the file optional. Since there is only one file to check for them, they don't have any perf impact.
For the first two lines, the brackets are delimiters to define the prefix before the glob pattern starts. Extracting this prefix (`../config/`) is needed to track resources correctly.
Lastly, you mention that brackets trigger a slow path, but that's the case only if your OS doesn't define `GLOG_BRACE`:
https://github.com/symfony/symfony/blob/8ffdef2958d4190994d7540013925b9580e03493/src/Symfony/Component/Config/Resource/GlobResource.php#L103
To my knowledge, only Solaris doesn't define it. If you're on Solaris, I'm sorry for you :)

Reply:
the condition before https://github.com/symfony/symfony/blob/8ffdef2958d4190994d7540013925b9580e03493/src/Symfony/Component/Config/Resource/GlobResource.php#L102 that line, bypass totally this path if there is a bracket in the pattern. So we fallback on the Finder.

Reply:
`defined('GLOB_BRACE')` is going to be true most of the time so we *do* enter this "if".
Aren't you looking for double stars instead? https://github.com/symfony/recipes/pull/678

Reply:
IIRC, it's skipped if there's a brace AND `GLOB_BRACE` isn't defined, which is the case on solaris according to @nicolas-grekas

Reply:
Wow, I'm on an alpine linux, and `GLOB_BRACE` is not defined. Hance the perf decrease.

Reply:
To be clear, I do have `GLOB_NOSORT` but not `GLOB_BRACE` on alpine. By removing the bracket, I go in the first if, then 0 is passed to the glob method. And everything works just fine. With a difference of ~100ms per request.

Reply:
confirmed no `GLOB_BRACE` on our alpine based prod images. great spot!

Reply:
OK, so I need to add musl libc to my list of platforms not supporting `GLOB_BRACE`. Too bad.
Maybe we could use `$container->import('../config/packages/*.yaml', 'glob', true);`.
Please give it a try and submit a PR if that works.
An alternative might be to use an implementation that doesn't require the Finder - ie we could have some logic that splits a single glob-with-braces into several of them. PR welcome as usual :)

Not a blocker for this very PR.

Reply:
Absolutely, I opened the discussion in a new ticket https://github.com/symfony/recipes/issues/712